packages/amplify-mqtt-server/src/client.ts

```diff
@@ -0,0 +1,440 @@
+import v4 from 'uuid';
+import * as retimer from 'retimer';
+import * as Connection from 'mqtt-connection';
+import * as steed from 'steed';
+
+import { MQTTServer } from './server';
+import { Logger } from 'pino';
+
+export class Client {
+  private logger: Logger;
+  private subscriptions = {};
+  nextId: number = 1;
+  inflight: {} = {};
+  inflightCounter: number = 0;
+  private keepAlive: number = 0;
+  private _lastDedupId: number = -1;
+  private _closed: boolean = false;
+  private _closing: boolean = false;
+  private timer;
+  private id;
+  private will;
+  private clean;
+  constructor(private connection: Connection, private server: MQTTServer) {
```

Comment:
Based on how the ```Client``` is getting instantiated, it will own the ```Connection``` instance, is everything cleaned up upon the destruction of the ```Client``` instance?

Reply:
We have cleanup code running which do its job